packages/notifi-react-card/lib/components/intercom/ChatMessageSection.tsx

```diff
@@ -1,40 +1,96 @@
-import clsx from 'clsx';
 import React from 'react';
+import { Virtuoso } from 'react-virtuoso';
 
-import { ChatMessageDate, ChatMessageDateProps } from './ChatMessageData';
+import { useIntercomChat } from '../../hooks/useIntercomChat';
+import { ChatMessageDate, ChatMessageDateProps } from './ChatMessageDate';
+import {
+  ChatWindowIntroSection,
+  ChatWindowIntroSectionProps,
+} from './ChatWindowIntroSection';
 import { MessageList, MessageListProps } from './MessageList';
 
 export type ChatMessageSectionProps = Readonly<{
   classNames?: Readonly<{
-    container: string;
-    content: string;
-    date: ChatMessageDateProps['classNames'];
+    chatWindowIntro: ChatWindowIntroSectionProps['classNames'];
     messageList: MessageListProps['classNames'];
+    date: ChatMessageDateProps['classNames'];
   }>;
   chatMessageSectionIntroContent?: string;
 }>;
 
 export const ChatMessageSection: React.FC<ChatMessageSectionProps> = ({
   classNames,
-  chatMessageSectionIntroContent = 'What can we help you with today?',
+  chatMessageSectionIntroContent,
 }) => {
+  const {
+    conversation,
+    setIsScrolling,
+    setVisibleRange,
+    setAtTop,
+    isLoading,
+    hasNextPage,
+  } = useIntercomChat({
+    conversationId: '8c5ccf41d93c4bb9b5e1e1014002e923',
+  });
+
   return (
-    <div
-      className={clsx(
-        'NotifiIntercomChatMessageSection__container',
-        classNames?.container,
+    <>
+      {conversation.feed.length === 0 ? (
+        <ChatWindowIntroSection
+          classNames={classNames?.chatWindowIntro}
+          startDate={conversation.createdDate}
+          chatMessageSectionIntroContent={chatMessageSectionIntroContent}
+        />
+      ) : null}
+
+      {conversation.feed.length === 0 ? null : (
+        <Virtuoso
+          atTopStateChange={setAtTop}
+          className="virtual-container"
+          data={conversation.feed}
+          followOutput="auto"
+          isScrolling={setIsScrolling}
+          rangeChanged={setVisibleRange}
+          itemContent={(index, feed) => {
+            return (
+              <div key={index}>
+                {isLoading ? (
+                  <div
+                    style={{
+                      display: 'flex',
+                      justifyContent: 'center',
+                      padding: '10px',
+                    }}
+                  >
+                    Loading...
+                  </div>
+                ) : null}
+                {index === 0 && !hasNextPage && (
```

Comment:
Should the Chat Window Intro Section always appear?

If it is at zero, it should always appear, no matter if it has a next page. The next page value in the API data tells us if there is more data than the pre-defined amount that we intended to display.

Reply:
Yeah, we should only show intro section when this is the very first message, but as we are fetching more messages each time the user scrolling, the first message always has been changed . But we don't want to show the intro section with the new fetched first message until it is the  actually very first message in the entire conversation list.

Reply:
Ah I think you are saying that if we are index === 0 and it's the last page of messages. 

Could we also make this clear? perhaps we can do something like `const IsFirstIndexOnLastPage = index === 0 && !hasNextPage`?